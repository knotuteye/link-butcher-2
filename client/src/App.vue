<template>
  <div id="app">
    <ImageBackground>
      <ColorOverlay>
        <div class="wrapper">
          <Branding name="pbid.io" subtitle="Link Shortener"></Branding>
          <URLInput />
          <Bubble
            v-if="newTuple"
            :short="newTuple.slug"
            :original="newTuple.url"
          />
          <RecentLinksBox>
            <RecentLink
              v-for="obj in recentLinks"
              :key="obj.slug"
              :short="obj.slug"
              :original="obj.url"
            >
            </RecentLink>
          </RecentLinksBox>
        </div>
      </ColorOverlay>
    </ImageBackground>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import ImageBackground from './components/ImageBackground.vue'
import ColorOverlay from './components/ColorOverlay.vue'
import Branding from './components/Branding.vue'
import URLInput from './components/URLInput.vue'
import RecentLinksBox from './components/RecentLinksBox.vue'
import RecentLink from './components/RecentLink.vue'
import Bubble from './components/Bubble.vue'
import { getRecentLinks } from './api/calls'

@Component({
  components: {
    ImageBackground,
    ColorOverlay,
    Branding,
    URLInput,
    RecentLinksBox,
    RecentLink,
    Bubble
  }
})
export default class App extends Vue {
  constructor() {
    super()
    this.fetchAndUpdateRecentLinks()
  }

  fetchAndUpdateRecentLinks() {
    getRecentLinks()
      .then(data => {
        this.$store.commit('updateRecentLinks', data)
      })
      .catch(() => {
        this.$store.commit('updateConnectionError', true)
      })
  }

  get recentLinks() {
    return this.$store.state.recentLinks
  }
  get newTuple() {
    return this.$store.state.newTuple
  }
}
</script>

<style>
* {
  margin: 0;
  outline: none;
}

#overlay > canvas {
  position: fixed;
  z-index: 1;
}

.overlay {
  display: flex;
  justify-content: center;
}

.wrapper {
  height: max-content;
  width: 80%;
  max-width: 40rem;
  margin-top: 25vh;
  z-index: 2;
  padding-bottom: 2rem;
  box-sizing: border-box;
}

.brand {
  margin-bottom: 5rem;
}

.url-input {
  margin-bottom: 5rem;
}
</style>
