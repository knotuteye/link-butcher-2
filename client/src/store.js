import Vue from 'vue'
import Vuex from 'vuex'
Vue.use(Vuex)
export default new Vuex.Store({
  state: {
    recentLinks: [],
    newTuple: null
  },
  mutations: {
    updateRecentLinks: function(state, arr) {
      state.recentLinks = arr
    }
  },
  actions: {},
  modules: {}
})
