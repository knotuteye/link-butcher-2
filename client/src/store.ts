import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    // server: 'http://localhost:5000/',
    server: window.location.href,
    recentLinks: new Array<{ slug: string; url: string }>(),
    newTuple: null,
    connectionError: false
  },
  mutations: {
    updateRecentLinks(state, arr) {
      state.recentLinks = arr
    },

    addToRecentLinkStack(state, tuple: { slug: string; url: string }) {
      const lastItem = JSON.parse(JSON.stringify(state.recentLinks))[0]

      if (lastItem.slug != tuple.slug) {
        state.recentLinks.unshift(tuple)
      }
    },

    updateNewLink(state, tuple) {
      state.newTuple = tuple
    },

    updateConnectionError(state, bool) {
      state.connectionError = bool
    }
  },
  actions: {},
  modules: {}
})
