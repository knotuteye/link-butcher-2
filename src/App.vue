<template>
  <div id="app">
    <ImageBackground>
      <ColorOverlay>
        <div class="wrapper">
          <Branding name="pbid.io" subtitle="Link Shortener"></Branding>
          <URLInput />
        </div>
      </ColorOverlay>
    </ImageBackground>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import ImageBackground from './components/ImageBackground.vue'
import ColorOverlay from './components/ColorOverlay.vue'
import Branding from './components/Branding.vue'
import URLInput from './components/URLInput.vue'
@Component({
  components: {
    ImageBackground,
    ColorOverlay,
    Branding,
    URLInput
  }
})
export default class App extends Vue {}
</script>

<style>
* {
  margin: 0;
  outline: none;
}

.overlay {
  display: flex;
  align-items: center;
}

.wrapper {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.brand {
  margin-bottom: 5rem;
}
</style>
