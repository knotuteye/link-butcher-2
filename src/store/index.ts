import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    recentLinks: [
      {
        short: 'xzcvgfds',
        original: 'https://www.google.com'
      },
      {
        short: 'r4e2sd6y',
        original: 'https://www.facebook.com'
      },
      {
        short: 'vlghoesa',
        original: 'https://www.techcrunch.com'
      },
      {
        short: 'rottieiw',
        original: 'https://web.whatsapp.com'
      }
    ]
  },
  mutations: {},
  actions: {},
  modules: {}
})
