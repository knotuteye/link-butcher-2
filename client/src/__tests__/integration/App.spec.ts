import { mount } from '@vue/test-utils'
import App from '../../App.vue'
import store from '../../store'

describe('App.vue', () => {
  test('is integrated as designed', () => {
    const wrapper = mount(App, { store: store })
    expect(wrapper.element).toMatchSnapshot()
  })
})
