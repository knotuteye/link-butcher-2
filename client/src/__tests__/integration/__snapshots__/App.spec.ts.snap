// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue is integrated as designed 1`] = `
<div
  id="app"
>
  <div
    class="background"
  >
    <div
      class="overlay"
      id="overlay"
    >
      <div
        class="wrapper"
      >
        <div
          class="brand slide-in-top"
          subtitle="Link Shortener"
        >
          <h1
            class="masthead"
          >
            
    pbid.io
  
          </h1>
        </div>
         
        <form
          class="url-input"
          novalidate="novalidate"
        >
          <input
            aria-label="Shrink the pasted link"
            type="submit"
            value="Shrink"
          />
           
          <input
            aria-label="Paste a link here"
            class=""
            name="url"
            placeholder="Paste a link to shorten it"
            required="required"
            spellcheck="false"
            type="url"
          />
        </form>
         
        <!---->
         
        <!---->
      </div>
    </div>
  </div>
</div>
`;
